core/src/main/scala/chisel3/package.scala

```diff
@@ -234,6 +235,117 @@ package object chisel3 {
+          // In the case of %f specifier there is a chance that we need more information - so capture till the 1st letter (a-zA-Z).
+          // Example cf"This is $val%2.2f here" - parts - Seq("This is ","%2.2f here") - the format specifier here is %2.2f.
+          val idx_of_fmt_str =
+            if (
+              !part.isEmpty() && part.charAt(0) == '%' && (part.size == 1 || (part.size >= 2 && part.charAt(1) != '%'))
+            ) part.indexWhere { _.isLetter }
+            else -1
+
+          // If no format specifier - pick default - %s
+
+          val fmt = if (idx_of_fmt_str >= 0) part.substring(0, idx_of_fmt_str + 1) else "%s"
+          val fmtArgs: Printable = arg match {
+            case b: Bits => {
+              require(fmt.size == 2, "In the case of bits, only single format char allowed!")
+              fmt match {
+                case "%s" => b.toPrintable
+                case "%n" => Name(b)
+                case "%N" => FullName(b)
+                // Default - let FirrtlFormat check validity of the format string to avoid repeating checks.
+                case f => FirrtlFormat(f.substring(1, 2), b)
+              }
+            }
+            case d: Data => {
+              require(
+                fmt == "%s" || fmt == "%n" || fmt == "%N",
+                "Non-bits only  allowed with (%s,%n, %N) format specifiers!"
+              )
+              fmt match {
+                case "%n" => Name(d)
+                case "%N" => FullName(d)
+                case "%s" => d.toPrintable
+                case x => {
+                  val msg = s"Illegal format specifier '$x'!\n"
+                  throw new UnknownFormatConversionException(msg)
+                }
+              }
+            }
+            case p: Printable => {
+              require(fmt == "%s", "Printables not allowed with format specifiers!")
+              p
+            }
+
+            // Generic case - use String.format (for example %d,%2.2f etc on regular Scala types)
+            // To support cases of %f format specifier on Int based types - need to manually cast for it to work.
+            case t => {
+              val castedT = (fmt.takeRight(1), t) match {
+                case ("f", v: Int) => v.asInstanceOf[Double]
+                case ("f", v: Short) => v.asInstanceOf[Double]
+                case ("f", v: Byte) => v.asInstanceOf[Double]
+                case ("f", v: Long) => v.asInstanceOf[Double]
+                case (_, t) => t
+              }
+
+              PString(String.format(fmt, castedT.asInstanceOf[AnyRef]))
+            }
+          }
+
+          // Remove format specifier from parts string
+          val modP = part.zipWithIndex.filter { _._2 > idx_of_fmt_str }.map { _._1 }.mkString
+
+          (modP, Some(fmtArgs))
+        }
+      }
+      // Combine the 1st part with the rest of the modified (format specifier removed) parts
+      val combParts = parts(0) +: partsAndSpecifierSeq.map { _._1 }
```

Comment:
```suggestion
      val combParts = parts(0) +: partsAndSpecifierSeq.map(_._1)
```
Just FYI, Scala style is to generally use `(...)` unless you need `{ ... }` which is only in cases when you need a `{ case ... => }` or when you need multiple statements in your anonymous function.